### PR TITLE
Price type queries tests

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Completes #2105 

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/data/queries/priceTypes/test/usePriceTypeForPrice.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		);
+
+		const { current: cachedDefaultPriceType } = result;
+
+		expect(cachedDefaultPriceType).toEqual(defaultPriceType);
+
+		expect(cachedDefaultPriceType.id).toEqual(defaultPriceType.id);
+
+		expect(cachedDefaultPriceType.name).toEqual(defaultPriceType.name);
+	});
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/data/queries/priceTypes/test/usePriceTypeForPrice.test.ts]: this is sooooo easy to read, great job 🏅

File: assets/prototype/domain/eventEditor/data/queries/priceTypes/useDefaultPriceType.ts [hunk 2]
```diff
@@ -1,15 +1,14 @@
 import isEmpty from 'ramda/src/isEmpty';
 import usePriceTypes from './usePriceTypes';
 import { PriceType } from '../../types';
+import { isDefaultPriceType } from '../../../../shared/predicates/priceTypes/selectionPredicates';
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/data/queries/priceTypes/useDefaultPriceType.ts]: YES! reusability FTW! 🎖

File: assets/prototype/domain/eventEditor/data/queries/priceTypes/useDefaultPriceType.ts [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
 
 /**
  * A custom react hook for retrieving the default price type object.
  */
 const useDefaultPriceType = (): PriceType => {
 	const allPriceTypes: PriceType[] = usePriceTypes();
-	const percentSurchargePriceTypes: PriceType[] = allPriceTypes.filter(
-		({ baseType, isPercent }) => baseType === 'SURCHARGE' && isPercent
-	);
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/data/queries/priceTypes/useDefaultPriceType.ts]: 😱  uhhhh... doesn't look like that was returning default prices !!!
was this caught by adding tests? If so then 👍 

[EDIT]
oops.. my bad... was thinking of prices not price types

File: assets/prototype/domain/eventEditor/data/queries/priceTypes/test/usePriceTypeForPrice.test.ts (hunk 1 above)
Comment (nicolad) [reply assets/prototype/domain/eventEditor/data/queries/priceTypes/test/usePriceTypeForPrice.test.ts]: agree, also I like the logic flow of the tests :medal_military:

File: assets/prototype/domain/eventEditor/data/queries/priceTypes/test/usePriceTypes.test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import usePriceTypes from '../usePriceTypes';
+import { ApolloMockedProvider } from '../../../../context';
+import { nodes } from './data';
+import useInitPriceTypeTestCache from './useInitPriceTypeTestCache';
+
+describe('usePriceTypes()', () => {
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/priceTypes/test/usePriceTypes.test.ts]: I love that this test file is so short and focused.
I feel guilty of having at least one test file that is too long, will need to split that somehow.

**Pull request merged**